tables/AlternateMans/test/tColumnarFile.cc

```diff
@@ -0,0 +1,164 @@
+#include <boost/test/unit_test.hpp>
+
+#include "../StokesIStMan/SimpleColumnarFile.h"
+#include "../StokesIStMan/BufferedColumnarFile.h"
+
+using casacore::VarBufferedColumnarFile;
+using casacore::SimpleColumnarFile;
+
+template<typename ColumnarFile>
+void TestEmptyConstructor() {
+  ColumnarFile file;
+  BOOST_CHECK_EQUAL(file.NRows(), 0);
+  BOOST_CHECK_EQUAL(file.Filename(), "");
+  BOOST_CHECK(!file.IsOpen());
+
+  file = ColumnarFile();
+  BOOST_CHECK_EQUAL(file.NRows(), 0);
+  BOOST_CHECK_EQUAL(file.Filename(), "");
+  BOOST_CHECK(!file.IsOpen());
+}
+
+template<typename ColumnarFile>
+void TestCreateAndOpen() {
+  constexpr size_t kStrideA = 10;
+  const std::string filename = "columnar_file_test.tmp";
+  {
+    ColumnarFile file = ColumnarFile::CreateNew(filename, {}, kStrideA);
+    BOOST_CHECK_EQUAL(file.Stride(), kStrideA);
+    BOOST_CHECK_EQUAL(file.NRows(), 0);
+    BOOST_CHECK_EQUAL(file.Filename(), filename);
+    BOOST_CHECK(file.IsOpen());
+  }
+
+  // See if re-creating works
+  constexpr size_t kStrideB = 20;
+  ColumnarFile file = ColumnarFile::CreateNew(filename, {}, kStrideB);
+  BOOST_CHECK_EQUAL(file.Stride(), kStrideB);
+  BOOST_CHECK_EQUAL(file.NRows(), 0);
+  BOOST_CHECK_EQUAL(file.Filename(), filename);
+  BOOST_CHECK(file.IsOpen());
+  file.AddRows(43);
+  BOOST_CHECK_EQUAL(file.NRows(), 43);
+  file.DeleteRow();
+  BOOST_CHECK_EQUAL(file.NRows(), 42);
+  BOOST_CHECK(file.IsOpen());
+  // Close by assigning to empty
+  file = ColumnarFile();
+  BOOST_CHECK(!file.IsOpen());
+
+  file = ColumnarFile::OpenExisting(filename, 0);
+  BOOST_CHECK_EQUAL(file.Stride(), kStrideB);
+  BOOST_CHECK_EQUAL(file.NRows(), 42);
+  BOOST_CHECK_EQUAL(file.Filename(), filename);
+  BOOST_CHECK(file.IsOpen());
+  file.AddRows(8);
+  BOOST_CHECK_EQUAL(file.NRows(), 50);
+  file.Close();
+  BOOST_CHECK(!file.IsOpen());
+
+  // Check if update worked
+  file = ColumnarFile::OpenExisting(filename, 0);
+  BOOST_CHECK_EQUAL(file.Stride(), kStrideB);
+  BOOST_CHECK_EQUAL(file.NRows(), 50);
+  file.Close();
+
+  unlink(filename.c_str());
+}
+
+template<typename ColumnarFile>
+void TestReadAndWrite() {
+  // Test a file with two columns: one is 14 complex floats, the other 3.
+  std::array<std::complex<float>, 14> data_a;
+  std::array<std::complex<float>, 3> data_b;
+  constexpr size_t kStride = sizeof(data_a) + sizeof(data_b);
+  const std::string filename = "columnar_file_test.tmp";
+  const std::array<unsigned char, 4> header = {1, 9, 8, 2};
+  ColumnarFile file = ColumnarFile::CreateNew(filename, header.size(), kStride);
+  file.WriteHeader(header.data());
+  BOOST_CHECK_EQUAL(file.Stride(), 17*8);
+
+  // Generate random data
+  for(size_t i=0; i!=std::size(data_a); ++i)
+      data_a[i] = std::complex<float>(i+3, i*2);
+  for(size_t i=0; i!=std::size(data_b); ++i)
+      data_b[i] = std::complex<float>(i*3+41, i*-2.0);
```

Comment:
I'm surprised casting a `std::array` to `std::size` just works. Still, using the array's `.size()` member is more straightforward:
```suggestion
  for(size_t i=0; i!=data_a.size(); ++i)
      data_a[i] = std::complex<float>(i+3, i*2);
  for(size_t i=0; i!=data_b.size(); ++i)
      data_b[i] = std::complex<float>(i*3+41, i*-2.0);
```

Reply:
size is not a type, it's a function (so it's not a cast). But I agree that .size() is nicer, I think I was using a regular c-array before, hence the std::size.